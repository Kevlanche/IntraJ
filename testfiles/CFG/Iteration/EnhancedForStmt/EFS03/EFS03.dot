digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[6]"}
{ rank = same; "TypeImportOnDemandDecl[4]"; "ClassDecl[7]"}
{ rank = same; "PackageAccess[5]"; "Modifiers[8]"; "Opt[11]"; "List[12]"; "List[13]"}
{ rank = same; "List[9]"; "MethodDecl[14]"}
{ rank = same; "Modifier[10]"; "Entry[15]"; "Modifiers[16]"; "PrimitiveTypeAccess[18]"; "List[19]"; "List[20]"; "Opt[21]"; "Exit[22]"}
{ rank = same; "List[17]"; "Block[22]"}
{ rank = same; "List[23]"}
{ rank = same; "VarDeclStmt[24]"; "EnhancedForStmt[44]"; "VarDeclStmt[85]"; "ExprStmt[93]"}
{ rank = same; "Modifiers[25]"; "ParTypeAccess[27]"; "List[31]"; "Modifiers[45]"; "TypeAccess[47]"; "VariableDeclarator[48]"; "VarAccess[51]"; "Block[52]"; "Modifiers[86]"; "PrimitiveTypeAccess[88]"; "List[89]"; "AssignSimpleExpr[94]"}
{ rank = same; "List[26]"; "TypeAccess[28]"; "List[29]"; "VariableDeclarator[32]"; "List[46]"; "List[49]"; "Opt[50]"; "List[53]"; "List[87]"; "VariableDeclarator[90]"; "VarAccess[95]"; "IntegerLiteral[96]"}
{ rank = same; "TypeAccess[30]"; "List[33]"; "Opt[34]"; "VarDeclStmt[54]"; "IfStmt[63]"; "VarDeclStmt[76]"; "List[91]"; "Opt[92]"}
{ rank = same; "ClassInstanceExpr[35]"; "Modifiers[55]"; "PrimitiveTypeAccess[57]"; "List[58]"; "Dot[64]"; "BreakStmt[72]"; "Opt[75]"; "Modifiers[77]"; "PrimitiveTypeAccess[79]"; "List[80]"}
{ rank = same; "ParTypeAccess[36]"; "List[40]"; "Opt[41]"; "List[42]"; "List[43]"; "List[56]"; "VariableDeclarator[59]"; "VarAccess[65]"; "MethodAccess[66]"; "List[74]"; "UncheckedExceptions[73]"; "List[78]"; "VariableDeclarator[81]"}
{ rank = same; "TypeAccess[37]"; "List[38]"; "List[60]"; "Opt[61]"; "List[68]"; "List[70]"; "List[71]"; "UncheckedExceptions[67]"; "List[82]"; "Opt[83]"}
{ rank = same; "TypeAccess[39]"; "IntegerLiteral[62]"; "StringLiteral[69]"; "IntegerLiteral[84]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeImportOnDemandDecl[4]" [label="TypeImportOnDemandDecl\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PackageAccess[5]" [label="PackageAccess\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[6]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[7]" [label="ClassDecl\n < EFS03 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[8]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[9]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[10]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[11]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[12]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[13]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[14]" [label="MethodDecl\n < m >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[15]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[16]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[17]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[18]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[19]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[20]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[21]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[22]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[23]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[24]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[25]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParTypeAccess[27]" [label="ParTypeAccess\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[28]" [label="TypeAccess\n < Collection >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[29]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[30]" [label="TypeAccess\n < String >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[31]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[32]" [label="VariableDeclarator\n < Collection<String> coll >", fillcolor=white   style=filled  ]
"List[33]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[34]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassInstanceExpr[35]" [label="ClassInstanceExpr\n <LinkedList>", fillcolor=white   style=filled  ]
"ParTypeAccess[36]" [label="ParTypeAccess\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[37]" [label="TypeAccess\n < LinkedList >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[38]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[39]" [label="TypeAccess\n < String >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[40]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[41]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[42]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[43]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"EnhancedForStmt[44]" [label="EnhancedForStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[45]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[46]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[47]" [label="TypeAccess\n < String >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[48]" [label="VariableDeclarator\n < String str >", fillcolor=white   style=filled  ]
"List[49]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[50]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[51]" [label="VarAccess\n < coll >", fillcolor=white   style=filled  ]
"Block[52]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[53]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[54]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[55]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[56]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[57]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[58]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[59]" [label="VariableDeclarator\n < int a >", fillcolor=white   style=filled  ]
"List[60]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[61]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[62]" [label="IntegerLiteral\n < 5 >", fillcolor=white   style=filled  ]
"IfStmt[63]" [label="IfStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[64]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarAccess[65]" [label="VarAccess\n < str >", fillcolor=white   style=filled  ]
"MethodAccess[66]" [label="MethodAccess\n < equals >", fillcolor=white   style=filled  ]
"List[68]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"StringLiteral[69]" [label="StringLiteral\n <  >", fillcolor=white   style=filled  ]
"List[70]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[71]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[67]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"BreakStmt[72]" [label="BreakStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[74]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[73]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[75]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VarDeclStmt[76]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[77]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[78]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[79]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[80]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[81]" [label="VariableDeclarator\n < int b >", fillcolor=white   style=filled  ]
"List[82]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[83]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[84]" [label="IntegerLiteral\n < 9 >", fillcolor=white   style=filled  ]
"VarDeclStmt[85]" [label="VarDeclStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[86]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[87]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[88]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[89]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"VariableDeclarator[90]" [label="VariableDeclarator\n < int c >", fillcolor=white   style=filled  ]
"List[91]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[92]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[93]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"AssignSimpleExpr[94]" [label="AssignSimpleExpr\n < c >", fillcolor=white   style=filled  ]
"VarAccess[95]" [label="VarAccess\n < c >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"IntegerLiteral[96]" [label="IntegerLiteral\n < 4 >", fillcolor=white   style=filled  ]
"Exit[22]" [label="Exit\n ", fillcolor=white   style=filled  ]
"TypeImportOnDemandDecl[4]" -> "PackageAccess[5]" [style=dashed, color=gray]
"List[3]" -> "TypeImportOnDemandDecl[4]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[9]" -> "Modifier[10]" [style=dashed, color=gray]
"Modifiers[8]" -> "List[9]" [style=dashed, color=gray]
"ClassDecl[7]" -> "Modifiers[8]" [style=dashed, color=gray]
"ClassDecl[7]" -> "Opt[11]" [style=dashed, color=gray]
"ClassDecl[7]" -> "List[12]" [style=dashed, color=gray]
"MethodDecl[14]" -> "Entry[15]" [style=dashed, color=gray]
"Modifiers[16]" -> "List[17]" [style=dashed, color=gray]
"MethodDecl[14]" -> "Modifiers[16]" [style=dashed, color=gray]
"MethodDecl[14]" -> "PrimitiveTypeAccess[18]" [style=dashed, color=gray]
"MethodDecl[14]" -> "List[19]" [style=dashed, color=gray]
"MethodDecl[14]" -> "List[20]" [style=dashed, color=gray]
"Modifiers[25]" -> "List[26]" [style=dashed, color=gray]
"VarDeclStmt[24]" -> "Modifiers[25]" [style=dashed, color=gray]
"ParTypeAccess[27]" -> "TypeAccess[28]" [style=dashed, color=gray]
"List[29]" -> "TypeAccess[30]" [style=dashed, color=gray]
"ParTypeAccess[27]" -> "List[29]" [style=dashed, color=gray]
"VarDeclStmt[24]" -> "ParTypeAccess[27]" [style=dashed, color=gray]
"VariableDeclarator[32]" -> "List[33]" [style=dashed, color=gray]
"ParTypeAccess[36]" -> "TypeAccess[37]" [style=dashed, color=gray]
"List[38]" -> "TypeAccess[39]" [style=dashed, color=gray]
"ParTypeAccess[36]" -> "List[38]" [style=dashed, color=gray]
"ClassInstanceExpr[35]" -> "ParTypeAccess[36]" [style=dashed, color=gray]
"ClassInstanceExpr[35]" -> "List[40]" [style=dashed, color=gray]
"ClassInstanceExpr[35]" -> "Opt[41]" [style=dashed, color=gray]
"ClassInstanceExpr[35]" -> "List[42]" [style=dashed, color=gray]
"ClassInstanceExpr[35]" -> "List[43]" [style=dashed, color=gray]
"Opt[34]" -> "ClassInstanceExpr[35]" [style=dashed, color=gray]
"VariableDeclarator[32]" -> "Opt[34]" [style=dashed, color=gray]
"List[31]" -> "VariableDeclarator[32]" [style=dashed, color=gray]
"VarDeclStmt[24]" -> "List[31]" [style=dashed, color=gray]
"List[23]" -> "VarDeclStmt[24]" [style=dashed, color=gray]
"Modifiers[45]" -> "List[46]" [style=dashed, color=gray]
"EnhancedForStmt[44]" -> "Modifiers[45]" [style=dashed, color=gray]
"EnhancedForStmt[44]" -> "TypeAccess[47]" [style=dashed, color=gray]
"VariableDeclarator[48]" -> "List[49]" [style=dashed, color=gray]
"VariableDeclarator[48]" -> "Opt[50]" [style=dashed, color=gray]
"EnhancedForStmt[44]" -> "VariableDeclarator[48]" [style=dashed, color=gray]
"EnhancedForStmt[44]" -> "VarAccess[51]" [style=dashed, color=gray]
"Modifiers[55]" -> "List[56]" [style=dashed, color=gray]
"VarDeclStmt[54]" -> "Modifiers[55]" [style=dashed, color=gray]
"VarDeclStmt[54]" -> "PrimitiveTypeAccess[57]" [style=dashed, color=gray]
"VariableDeclarator[59]" -> "List[60]" [style=dashed, color=gray]
"Opt[61]" -> "IntegerLiteral[62]" [style=dashed, color=gray]
"VariableDeclarator[59]" -> "Opt[61]" [style=dashed, color=gray]
"List[58]" -> "VariableDeclarator[59]" [style=dashed, color=gray]
"VarDeclStmt[54]" -> "List[58]" [style=dashed, color=gray]
"List[53]" -> "VarDeclStmt[54]" [style=dashed, color=gray]
"Dot[64]" -> "VarAccess[65]" [style=dashed, color=gray]
"List[68]" -> "StringLiteral[69]" [style=dashed, color=gray]
"MethodAccess[66]" -> "List[68]" [style=dashed, color=gray]
"MethodAccess[66]" -> "List[70]" [style=dashed, color=gray]
"MethodAccess[66]" -> "List[71]" [style=dashed, color=gray]
"MethodAccess[66]" -> "UncheckedExceptions[67]" [style=dashed, color=gray]
"Dot[64]" -> "MethodAccess[66]" [style=dashed, color=gray]
"IfStmt[63]" -> "Dot[64]" [style=dashed, color=gray]
"BreakStmt[72]" -> "List[74]" [style=dashed, color=gray]
"BreakStmt[72]" -> "UncheckedExceptions[73]" [style=dashed, color=gray]
"IfStmt[63]" -> "BreakStmt[72]" [style=dashed, color=gray]
"IfStmt[63]" -> "Opt[75]" [style=dashed, color=gray]
"List[53]" -> "IfStmt[63]" [style=dashed, color=gray]
"Modifiers[77]" -> "List[78]" [style=dashed, color=gray]
"VarDeclStmt[76]" -> "Modifiers[77]" [style=dashed, color=gray]
"VarDeclStmt[76]" -> "PrimitiveTypeAccess[79]" [style=dashed, color=gray]
"VariableDeclarator[81]" -> "List[82]" [style=dashed, color=gray]
"Opt[83]" -> "IntegerLiteral[84]" [style=dashed, color=gray]
"VariableDeclarator[81]" -> "Opt[83]" [style=dashed, color=gray]
"List[80]" -> "VariableDeclarator[81]" [style=dashed, color=gray]
"VarDeclStmt[76]" -> "List[80]" [style=dashed, color=gray]
"List[53]" -> "VarDeclStmt[76]" [style=dashed, color=gray]
"Block[52]" -> "List[53]" [style=dashed, color=gray]
"EnhancedForStmt[44]" -> "Block[52]" [style=dashed, color=gray]
"List[23]" -> "EnhancedForStmt[44]" [style=dashed, color=gray]
"Modifiers[86]" -> "List[87]" [style=dashed, color=gray]
"VarDeclStmt[85]" -> "Modifiers[86]" [style=dashed, color=gray]
"VarDeclStmt[85]" -> "PrimitiveTypeAccess[88]" [style=dashed, color=gray]
"VariableDeclarator[90]" -> "List[91]" [style=dashed, color=gray]
"VariableDeclarator[90]" -> "Opt[92]" [style=dashed, color=gray]
"List[89]" -> "VariableDeclarator[90]" [style=dashed, color=gray]
"VarDeclStmt[85]" -> "List[89]" [style=dashed, color=gray]
"List[23]" -> "VarDeclStmt[85]" [style=dashed, color=gray]
"AssignSimpleExpr[94]" -> "VarAccess[95]" [style=dashed, color=gray]
"AssignSimpleExpr[94]" -> "IntegerLiteral[96]" [style=dashed, color=gray]
"ExprStmt[93]" -> "AssignSimpleExpr[94]" [style=dashed, color=gray]
"List[23]" -> "ExprStmt[93]" [style=dashed, color=gray]
"Block[22]" -> "List[23]" [style=dashed, color=gray]
"Opt[21]" -> "Block[22]" [style=dashed, color=gray]
"MethodDecl[14]" -> "Opt[21]" [style=dashed, color=gray]
"MethodDecl[14]" -> "Exit[22]" [style=dashed, color=gray]
"List[13]" -> "MethodDecl[14]" [style=dashed, color=gray]
"ClassDecl[7]" -> "List[13]" [style=dashed, color=gray]
"List[6]" -> "ClassDecl[7]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[6]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[22]" -> "AssignSimpleExpr[94]" [color=red, constraint=false]
"AssignSimpleExpr[94]" -> "Exit[22]" [color=blue, constraint=false, xlabel=" "] 
"AssignSimpleExpr[94]" -> "IntegerLiteral[96]" [color=red, constraint=false]
"IntegerLiteral[96]" -> "AssignSimpleExpr[94]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[96]" -> "VariableDeclarator[90]" [color=red, constraint=false]
"VariableDeclarator[90]" -> "IntegerLiteral[96]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[90]" -> "VarAccess[51]" [color=red, constraint=false]
"VariableDeclarator[90]" -> "MethodAccess[66]" [color=red, constraint=false]
"VariableDeclarator[81]" -> "VarAccess[51]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[81]" -> "IntegerLiteral[84]" [color=red, constraint=false]
"IntegerLiteral[84]" -> "VariableDeclarator[81]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[84]" -> "MethodAccess[66]" [color=red, constraint=false]
"MethodAccess[66]" -> "VariableDeclarator[90]" [color=blue, constraint=false, xlabel="T "] 
"MethodAccess[66]" -> "IntegerLiteral[84]" [color=blue, constraint=false, xlabel="F "] 
"MethodAccess[66]" -> "StringLiteral[69]" [color=red, constraint=false]
"StringLiteral[69]" -> "MethodAccess[66]" [color=blue, constraint=false, xlabel=" "] 
"StringLiteral[69]" -> "VarAccess[65]" [color=red, constraint=false]
"VarAccess[65]" -> "StringLiteral[69]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[65]" -> "VariableDeclarator[59]" [color=red, constraint=false]
"VariableDeclarator[59]" -> "VarAccess[65]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[59]" -> "IntegerLiteral[62]" [color=red, constraint=false]
"IntegerLiteral[62]" -> "VariableDeclarator[59]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[62]" -> "VariableDeclarator[48]" [color=red, constraint=false]
"VariableDeclarator[48]" -> "IntegerLiteral[62]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[48]" -> "VarAccess[51]" [color=red, constraint=false]
"VarAccess[51]" -> "VariableDeclarator[48]" [color=blue, constraint=false, xlabel="T "] 
"VarAccess[51]" -> "VariableDeclarator[90]" [color=blue, constraint=false, xlabel="F "] 
"VarAccess[51]" -> "VariableDeclarator[32]" [color=red, constraint=false]
"VarAccess[51]" -> "VariableDeclarator[81]" [color=red, constraint=false]
"VariableDeclarator[32]" -> "VarAccess[51]" [color=blue, constraint=false, xlabel=" "] 
"VariableDeclarator[32]" -> "ClassInstanceExpr[35]" [color=red, constraint=false]
"ClassInstanceExpr[35]" -> "VariableDeclarator[32]" [color=blue, constraint=false, xlabel=" "] 
"ClassInstanceExpr[35]" -> "Entry[15]" [color=red, constraint=false]
"Entry[15]" -> "ClassInstanceExpr[35]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[6]" [style=invis]
"Modifiers[8]" -> "Opt[11]" -> "List[12]" -> "List[13]" [style=invis]
"Entry[15]" -> "Modifiers[16]" -> "PrimitiveTypeAccess[18]" -> "List[19]" -> "List[20]" -> "Opt[21]" -> "Exit[22]" [style=invis]
"VarDeclStmt[24]" -> "EnhancedForStmt[44]" -> "VarDeclStmt[85]" -> "ExprStmt[93]" [style=invis]
"Modifiers[25]" -> "ParTypeAccess[27]" -> "List[31]" [style=invis]
"TypeAccess[28]" -> "List[29]" [style=invis]
"List[33]" -> "Opt[34]" [style=invis]
"ParTypeAccess[36]" -> "List[40]" -> "Opt[41]" [style=invis]
"TypeAccess[37]" -> "List[38]" [style=invis]
"Modifiers[45]" -> "TypeAccess[47]" -> "VariableDeclarator[48]" -> "VarAccess[51]" -> "Block[52]" [style=invis]
"List[49]" -> "Opt[50]" [style=invis]
"VarDeclStmt[54]" -> "IfStmt[63]" -> "VarDeclStmt[76]" [style=invis]
"Modifiers[55]" -> "PrimitiveTypeAccess[57]" -> "List[58]" [style=invis]
"List[60]" -> "Opt[61]" [style=invis]
"Dot[64]" -> "BreakStmt[72]" -> "Opt[75]" [style=invis]
"VarAccess[65]" -> "MethodAccess[66]" [style=invis]
"Modifiers[77]" -> "PrimitiveTypeAccess[79]" -> "List[80]" [style=invis]
"List[82]" -> "Opt[83]" [style=invis]
"Modifiers[86]" -> "PrimitiveTypeAccess[88]" -> "List[89]" [style=invis]
"List[91]" -> "Opt[92]" [style=invis]
"VarAccess[95]" -> "IntegerLiteral[96]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
98", fillcolor=1]
Edges [label="Total Nr. Edges
131", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
16", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
17", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}