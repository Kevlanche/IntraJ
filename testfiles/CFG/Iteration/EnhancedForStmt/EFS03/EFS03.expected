NodeName: Entry[15] Successors: [ClassInstanceExpr[35]]
NodeName: ClassInstanceExpr[35] Successors: [VariableDeclarator[32]]
NodeName: VariableDeclarator[32] Successors: [VarAccess[51]]
NodeName: VarAccess[51] Successors: [VariableDeclarator[48], VariableDeclarator[90]]
NodeName: VariableDeclarator[48] Successors: [IntegerLiteral[62]]
NodeName: IntegerLiteral[62] Successors: [VariableDeclarator[59]]
NodeName: VariableDeclarator[59] Successors: [VarAccess[65]]
NodeName: VarAccess[65] Successors: [StringLiteral[69]]
NodeName: StringLiteral[69] Successors: [MethodAccess[66]]
NodeName: MethodAccess[66] Successors: [VariableDeclarator[90], IntegerLiteral[84]]
NodeName: VariableDeclarator[90] Successors: [IntegerLiteral[96]]
NodeName: IntegerLiteral[96] Successors: [AssignSimpleExpr[94]]
NodeName: AssignSimpleExpr[94] Successors: [Exit[22]]
NodeName: Exit[22] Successors: []
NodeName: IntegerLiteral[84] Successors: [VariableDeclarator[81]]
NodeName: VariableDeclarator[81] Successors: [VarAccess[51]]

NodeName: Exit[22] Predecessor: [AssignSimpleExpr[94]]
NodeName: AssignSimpleExpr[94] Predecessor: [IntegerLiteral[96]]
NodeName: IntegerLiteral[96] Predecessor: [VariableDeclarator[90]]
NodeName: VariableDeclarator[90] Predecessor: [VarAccess[51], MethodAccess[66]]
NodeName: VarAccess[51] Predecessor: [VariableDeclarator[32], VariableDeclarator[81]]
NodeName: VariableDeclarator[32] Predecessor: [ClassInstanceExpr[35]]
NodeName: ClassInstanceExpr[35] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: VariableDeclarator[81] Predecessor: [IntegerLiteral[84]]
NodeName: IntegerLiteral[84] Predecessor: [MethodAccess[66]]
NodeName: MethodAccess[66] Predecessor: [StringLiteral[69]]
NodeName: StringLiteral[69] Predecessor: [VarAccess[65]]
NodeName: VarAccess[65] Predecessor: [VariableDeclarator[59]]
NodeName: VariableDeclarator[59] Predecessor: [IntegerLiteral[62]]
NodeName: IntegerLiteral[62] Predecessor: [VariableDeclarator[48]]
NodeName: VariableDeclarator[48] Predecessor: [VarAccess[51]]
